crates/hyperswitch_connectors/src/connectors/nexixpay/transformers.rs

```diff
@@ -1,80 +1,420 @@
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct Recurrence {
+    action: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexixpayPaymentsResponse {
+    operation: Operation,
+    three_d_s_auth_request: String,
+    three_d_s_auth_url: Secret<url::Url>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ThreeDSAuthResult {
+    authentication_value: Secret<String>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+pub enum NexixpayPaymentIntent {
+    Capture,
+    Cancel,
+    Authorize,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexixpayConnectorMetaData {
+    pub three_d_s_auth_result: Option<ThreeDSAuthResult>,
+    pub three_d_s_auth_response: Option<String>,
+    pub authorization_operation_id: Option<String>,
+    pub capture_operation_id: Option<String>,
+    pub cancel_operation_id: Option<String>,
+    pub psync_flow: NexixpayPaymentIntent,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct UpdateNexixpayConnectorMetaData {
+    pub three_d_s_auth_result: Option<ThreeDSAuthResult>,
+    pub three_d_s_auth_response: Option<String>,
+    pub authorization_operation_id: Option<String>,
+    pub capture_operation_id: Option<String>,
+    pub cancel_operation_id: Option<String>,
+    pub psync_flow: Option<NexixpayPaymentIntent>,
+    pub meta_data: serde_json::Value,
+    pub is_auto_capture: bool,
+}
+
+fn update_nexi_meta_data(
+    update_request: UpdateNexixpayConnectorMetaData,
+) -> CustomResult<serde_json::Value, errors::ConnectorError> {
+    let nexixpay_meta_data =
+        serde_json::from_value::<NexixpayConnectorMetaData>(update_request.meta_data)
+            .change_context(errors::ConnectorError::ParsingFailed)?;
+
+    Ok(serde_json::json!(NexixpayConnectorMetaData {
+        three_d_s_auth_result: update_request
+            .three_d_s_auth_result
+            .or(nexixpay_meta_data.three_d_s_auth_result),
+        three_d_s_auth_response: update_request
+            .three_d_s_auth_response
+            .or(nexixpay_meta_data.three_d_s_auth_response),
+        authorization_operation_id: update_request
+            .authorization_operation_id
+            .clone()
+            .or(nexixpay_meta_data.authorization_operation_id.clone()),
+        capture_operation_id: {
+            if update_request.is_auto_capture {
+                update_request
+                    .authorization_operation_id
+                    .clone()
+                    .or(nexixpay_meta_data.authorization_operation_id)
+            } else {
+                update_request.capture_operation_id
+            }
+            .or(nexixpay_meta_data.capture_operation_id)
+        },
+        cancel_operation_id: update_request
+            .cancel_operation_id
```

Comment:
Since this function is generic , there is a chance that we may rotate the keys. So check the keys in metadata first ,if not present then update it from update_request. Please change it for other keys as well.
```suggestion
        cancel_operation_id: nexixpay_meta_data.cancel_operation_id.or(update_request
            .cancel_operation_id),
```

Reply:
this create problem in partial captures. You will get new captureId and you will not be updating that if done in this way.